src/java/com/unifina/security/RedirectAppendingAuthenticationEntryPoint.java

```diff
@@ -36,13 +40,39 @@ protected String buildRedirectUrlToLoginPage(HttpServletRequest request, HttpSer
 				|| request.getRequestURI().equals(getFullURI(defaultRedirectURI))) {
 			return loginPageUrl;
 		} else if (LoginRedirectValidator.isValid(request.getRequestURL().toString())){
-			return loginPageUrl + "?" + REDIRECT_PARAM_NAME + "=" + request.getRequestURL().toString();
+			final String value = String.format("%s?%s", request.getRequestURL().toString(), params(request));
+			final String url = String.format("%s?%s=%s", loginPageUrl, REDIRECT_PARAM_NAME, urlencode(value));
+			return url;
 		} else {
 			log.warn("Redirect url rejected: "+request.getRequestURL());
 			return loginPageUrl;
 		}
 	}
 
+	private String params(final HttpServletRequest request) {
```

Comment:
Do you really need to do this yourself? What about [HttpServletRequest#getQueryString()](https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletRequest.html#getQueryString())?